Code/Tools/FBuild/FBuildCore/Graph/DependencyListNode.cpp

```diff
@@ -0,0 +1,243 @@
+    bucket_t Visiteds[ iBuckets ];
+
+    const Array< AString > & Patterns;
+
+    explicit DependencyListBuilder( const Array< AString > & patterns )
+        : Patterns( patterns )
+    {
+    }
+
+    DependencyListBuilder( const DependencyListBuilder& ) = delete;
+    DependencyListBuilder& operator =( const DependencyListBuilder& ) = delete;
+
+    bool FindOrAddNode( const Node * node )
+    {
+        ASSERT( node );
+
+        bucket_t & bucket = Visiteds[ node->GetNameCRC() & mBuckets ];
+
+        if ( bucket.Find( node ) == nullptr )
+        {
+            bucket.Append( node );
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    void Collect( const Node * node )
+    {
+        if ( FindOrAddNode( node ) )
+        {
+            if ( node->GetType() == Node::ALIAS_NODE )
+            {
+                // resolve aliases to real nodes
+                AliasNode * aliasNode = node->CastTo< AliasNode >();
+                Collect( aliasNode->GetAliasedNodes() );
+            }
+
+            Collect( node->GetStaticDependencies() );
+            Collect( node->GetDynamicDependencies() );
+        }
+    }
+
+    void Collect( const Dependencies & deps )
+    {
+        for ( const Dependency & dep : deps )
+        {
+            if ( dep.IsWeak() == false )
+            {
+                Collect( dep.GetNode() );
+            }
+        }
+    }
+
+    bool IsMatch( const char * filename ) const
+    {
+        for ( const AString & pattern : Patterns )
+        {
+            if ( PathUtils::IsWildcardMatch( pattern.Get(), filename ) )
+            {
+                return true;
+            }
+        }
+        return Patterns.IsEmpty();
+    }
+
+    bool DumpTxtFile( const char* filename ) const
+    {
+        ASSERT( filename );
+
+        // Collect all file dependencies
+
+        Array< const AString * > dependencyList;
+
+        for ( const bucket_t & bucket : Visiteds )
+        {
+            for ( const Node * node : bucket )
+            {
+                if ( node->IsAFile() && IsMatch( node->GetName().Get() ) )
+                {
+                    dependencyList.Append( &node->GetName() );
+                }
+            }
+        }
+
+        // Sort collected filenames
+
+        dependencyList.Sort([]( const AString * a, const AString * b ) {
+            return ( a->Compare( *b ) < 0 );
+        });
+
+        // Prepare output buffer
+
+        AStackString<> buffer;
+
+        for ( const AString * depname : dependencyList )
+        {
+            buffer += *depname;
+            buffer += "\r\n";
+        }
+
+        // Dump to txt file
+
+        FileStream dmp;
+        if ( dmp.Open( filename, FileStream::WRITE_ONLY ) == false )
+        {
+            return false;
+        }
+        else
+        {
+            dmp.WriteBuffer( buffer.Get(), buffer.GetLength() * sizeof(*buffer.Get()) );
+            dmp.Close();
```

Comment:
We should check if the write succeeded here by comparing the return result to the size.
Though ok, you don't need to multiply the GetLength() by the size of a char.
The close can also be omitted (FileStream is self-closing). You only need to call Close() if you explicitly control the point at which the file is closed.